lib/core/uploaderError.js

```diff
@@ -17,29 +17,43 @@
 
 var sundial = require('sundial');
 
-// Create a new object, that prototypally inherits from the Error constructor.
-function UploaderError(message, stage, error) {
-
-  this.name = this.constructor.name;
-  this.message = message;
+// if another error is passed in then attach it
```

Comment:
I worry that this has the potential to be confusing. I really don't like the way the redundancy shows up in the server logs (nor the formatting? is there anything to be done about the insane proliferation of `\`?? They multiply like rabbits when there's more than one level of error)